### ISSUE TITLE
`IMessageSink` can't be injected into fixtures in xUnit.net v3

### ISSUE THREAD

**Issue by 0xced** [xunit/xunit#3001]
While it was previously fine to inject `IMessageSink` into fixtures, it doesn't work anymore with xUnit.net v3 (0.2.0-pre.69).

From the What’s New in v3 document:

> Several classes have had their constructors simplified by removing `IMessageSink` parameters that were previously used to send diagnostic messages to. Instead, developers can use the ambient `TestContext.Current.SendDiagnosticMessage` to simplify the sending of diagnostic messages.

I think it should not impact fixtures. If you're developing a fixture which references only `xunit.v3.extensibility.core` then you don't have access to `TestContext.Current.SendDiagnosticMessage`. Note: this is a real scenario, see https://github.com/testcontainers/testcontainers-dotnet/pull/1165

This is the exception I get when running tests in Rider:

```
Xunit.Sdk.TestPipelineException
Class fixture type 'XunitFixtureSink.SampleFixture' had one or more unresolved constructor arguments: IMessageSink messageSink
   at Xunit.v3.FixtureMappingManager.GetFixture(Type fixtureType) in D:\a\xunit\xunit\src\xunit.v3.core\Utility\FixtureMappingManager.cs:line 166
   at Xunit.v3.FixtureMappingManager.InitializeAsync(IReadOnlyCollection`1 fixtureTypes) in D:\a\xunit\xunit\src\xunit.v3.core\Utility\FixtureMappingManager.cs:line 226
   at Xunit.v3.ExceptionAggregator.RunAsync(Func`1 code) in D:\a\xunit\xunit\src\xunit.v3.core\Exceptions\ExceptionAggregator.cs:line 120
```

Oddly enough, I don't get this exception when running `dotnet test` but sending a diagnostic message with `messageSink.OnMessage(new DiagnosticMessage("[…]"))` doesn't print this diagnostic message.

Full reproduction steps are available in my [XunitFixtureSink](https://github.com/0xced/XunitFixtureSink) repository.

**Comment by bradwilson** [comment]
> While it was previously fine to inject `IMessageSink` into fixtures, it doesn't work anymore with xUnit.net v3 (0.2.0-pre.69).

I don't think this was intentional on my part, so I'll see what happened.

> If you're developing a fixture which references only `xunit.v3.extensibility.core` then you don't have access to `TestContext.Current.SendDiagnosticMessage`.

You should have access to it, as it lives in `xunit.v3.core`.

> sending a diagnostic message with `messageSink.OnMessage(new DiagnosticMessage("[…]"))` doesn't print this diagnostic message.

`dotnet test` hides out output by default, including showing diagnostic messages. After [enabling diagnostic messages](https://xunit.net/docs/configuration-files#diagnosticMessages) in your config file, you'll also need to pass `--logger "console;verbosity=normal` to your `dotnet test` invocation to see our output.

**Comment by bradwilson** [comment]
> `dotnet` test hides our output by default, including showing diagnostic messages.

I take it back, we report diagnostic messages as warnings to VSTest, so they get shown even with the default minimal output.

**Comment by bradwilson** [comment]
Fix available in Core Framework v3 `0.3.0-pre.8` https://xunit.net/docs/using-ci-builds

Turns out the issue is conditional. We don't normally have a diagnostic message sink when diagnostic messages aren't enabled, so you'd only see this when diagnostic messages are disabled (and we run all our tests with diagnostic messages enabled), so the fix is to ensure that we always inject a message sink even when diagnostic messages aren't enabled.
